coding/python.md

```diff
@@ -0,0 +1,21 @@
+---
+layout: base
+---
+
+# Python code standards
+
+This document outlines the rules for writing Python code across all our Projects.
+
+## Code style and formatting
+
+All code must conform to [PEP 8](https://pep8.org/) and be formatted with [black](https://github.com/ambv/black) using the option `--line-length 79`.
+
+Project should be configured to block merging on pull requests that do not conform to these standards.
+
+## Testing
+
+Although tests are lacking on many of our existing projects, writing tests is strongly encouraged. Please write tests for any new code if at all possible, and any efforts to improve test coverage on any project would be greatly appreciated.
```

Comment:
Only strongly encouraged? Shouldn't tests be mandatory?

Reply:
> And unfortunately we're not in a place at the moment to enforce 100% test coverage on all new work.

Why's that?

Reply:
I responded in possibly too much depth over at https://github.com/canonical-webteam/practices/issues/122#issuecomment-441273018.

I didn't want to start that discussion as part of this PR because I think it's better to initially document current practice, so we can build up the practices faster, but I'm very happy to explore evolving the policy.